src/lib/core/events.ts

```diff
@@ -0,0 +1,42 @@
+import events from "events";
+
+const COMMAND_PREFIX = "request";
+
+class EventEmitter extends events.EventEmitter {
+
+  public oncePromised(event: string | symbol, listener: (...args: any[]) => void = () => undefined) {
+    return new Promise<any[]>((resolve) => {
+      this.once(event, (...args: any[]) => {
+        listener(...args);
+        resolve(args);
+      });
+    });
+  }
+
+  public onPromised(event: string | symbol, listener: (...args: any[]) => void = () => undefined) {
+    return new Promise<any[]>((resolve) => {
+      this.on(event, (...args: any[]) => {
```

Comment:
I'm not 100% if this will do the trick. As you probably know, promises are one-off operations. Once resolved or rejected, their done. 

This means for multiple emissions, like WS-based events, you'll have to have multiple `onPromised()` calls. This is where Observables offer a much better abstraction (and can always downgraded to a promise if needed).

Reply:
Yep that will only works for the first event, I am fine with observable,  will try and see how it looks